packages/shared/src/errors/createErrorTypeGuard.ts

```diff
@@ -29,6 +29,14 @@ export function createErrorTypeGuard<T extends new (...args: any[]) => Value>(
     if (!target) {
       throw new TypeError(`${ErrorClass.kind || ErrorClass.name} type guard requires an error object`);
     }
+    // Use duck-typing with 'kind' property to handle cross-bundle scenarios
+    // where instanceof fails due to different class instances
+    if (ErrorClass.kind && typeof target === 'object' && target !== null && 'constructor' in target) {
+      const targetConstructor = (target as { constructor?: { kind?: string } }).constructor;
+      if (targetConstructor?.kind === ErrorClass.kind) {
+        return true;
+      }
+    }
```

Comment:
The problem encountered is:
- clerk-js throws ClerkAPIResponseError instances
- @clerk/elements uses isClerkAPIResponseError() to check errors
- Both import from @clerk/shared, but due to bundling, they get different copies of the ClerkAPIResponseError class
- instanceof fails across different class instances